crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2852,6 +2852,35 @@ impl Room {
         Ok(Invite { invitee, inviter })
     }
 
+    /// Get the membership details for the current user.
+    ///
+    /// Returns:
+    ///     - If the current user was present in the room, a tuple of the
+    ///       current user's [`RoomMember`](RoomMember) info and the member info
+    ///       of the sender of that member event.
+    ///     - If the current user is not present, an error.
+    pub async fn own_membership_details(&self) -> Result<(RoomMember, Option<RoomMember>)> {
```

Comment:
Can we get a simple smoke test for this method, I know it's a pretty simple method but it would still be nice to test its basic functionality.